### DISCUSSION TITLE
Postgres dialect doubles percentage sign for column labeled as "%"

### DISCUSSION THREAD

**Discussion by rafbgarcia** [sqlalchemy/sqlalchemy#8077] [Usage Questions] [answered]
### Describe the bug

I'm not sure this is a bug.

The postgres dialect doubles the percentage sign on column aliased as `"%"`.

### To Reproduce

```python
from sqlalchemy import column, select
from sqlalchemy.dialects import postgresql

query = select(column("percentage").label("%"))
print(str(query.compile()))
print(str(query.compile(dialect=postgresql.dialect())))
```

### Error

```
SELECT percentage AS "%"
SELECT percentage AS "%%"
```

### Versions

- Python: 3.9
- SQLAlchemy: 1.4.36

### Additional context

Is this a bug? If not, how to get around it?

**Comment by CaselIT** [comment, marked as answer]
Hi,

That's the expected behaviour since the default param style for postgress uses the format version that requires escaping of %.

This should not be a problem in general, unless you are somehow executing the sql outside sqlalchemy

**Reply by rafbgarcia** [reply]
Hi @CaselIT, that makes sense.

I need to be able to label columns as "%", do you know how can I achieve that using Postgres dialect?

**Reply by CaselIT** [reply]
That is (should be) supported.
Do you have a stacktrace or an error of sime kind?

**Reply by rafbgarcia** [reply]
@CaselIT there is no error, it just outputs doubled "%" as shown in the `Error` section above.

**Reply by rafbgarcia** [reply]
<img width="588" alt="Untitled_-_Jupyter_Notebook" src="https://user-images.githubusercontent.com/1904314/171039210-a7c72912-b72f-47a4-b2fd-cf9d14fc831e.png">

**Reply by CaselIT** [reply]
I understood that, but I'm not sure why is that an issue.
Do you have an error executing that statement with sqlalchemy?

**Reply by rafbgarcia** [reply]
No. The issue is that the label name comes as "%%" instead of "%" and that's incorrect.

The alias should be `%` and not `%%`.

I need to get the label as is, just like I would by labeling with any other name:

<img width="741" alt="Untitled_-_Jupyter_Notebook" src="https://user-images.githubusercontent.com/1904314/171043466-7747c6b2-028a-485f-91dc-cec42d71bac1.png">

**Reply by CaselIT** [reply]
> No. The issue is that the label name comes as "%%" instead of "%" and that's incorrect.

Again, postgresql uses by default the format style for params that requires escaping of `%` chars, so that's working as expected.

Running that statement using sqlalchemy returns the expected `[{'%': 'something'}]`.
```py
epg = create_engine("postgresql://scott:tiger@localhost/test", echo=True)
s = sa.select(sa.literal('something').label('%'))
with epg.connect() as conn:
    r = conn.execute(s)
    print(r.mappings().all())
```

Are you using the SQL statement directly? You can change the param style by setting `paramstyle` when instantiating the dialect. The supported values are specified in this documentation https://docs.sqlalchemy.org/en/14/core/engines.html?highlight=paramstyle#sqlalchemy.create_engine.params.paramstyle

**Reply by rafbgarcia** [reply]
Yes, I need the SQL directly to query another data source that's based on Postgres. I think the `paramstyle` is what I need, thank you!
